uma-sdk/src/commonMain/kotlin/me/uma/protocol/Currency.kt

```diff
@@ -43,11 +39,101 @@ data class Currency(
      *
      * For details on edge cases and examples, see https://github.com/uma-universal-money-address/protocol/blob/main/umad-04-lnurlp-response.md.
      */
-    val decimals: Int,
-) {
-    fun toJson() = serialFormat.encodeToString(this)
+    val decimals: Int
 }
 
+/**
+ * Creates a [Currency] which contains information about currencies that the VASP is able to receive.
+ *
+ * @param code The currency code, eg. "USD".
+ * @param name The full currency name, eg. "US Dollars".
+ * @param symbol The symbol of the currency, eg. "$".
+ * @param millisatoshiPerUnit The conversion rate from the smallest unit of the currency to millisatoshis.
+ * @param decimals The number of decimal places in the currency.
+ * @param minSendable Minimum amount that can be sent in this currency. This is in the smallest unit of the
+ *      currency (eg. cents for USD).
+ * @param maxSendable Maximum amount that can be sent in this currency. This is in the smallest unit of the
+ *      currency (eg. cents for USD).
+ * @param senderUmaVersion The UMA version of the sender VASP. This information can be obtained from the
+ *      [LnurlpRequest].
+ * @return the [Currency] to be sent to the sender VASP.
+ */
+@JvmOverloads
+fun getCurrency(
```

Comment:
nit (although I know I'm bad about this in a few places):

```suggestion
fun createCurrency(
```